### PR TITLE
Add GetFieldParams and GetFieldParamsWithOptions functions

### PR DESCRIPTION
**Pull request by saturn4er**
Here's example implementation for #260, not really sure about names I used :)

### CODE REVIEW

**Comment by saturn4er** [comment]
@caarlos0 Any problems with this concept?

**Comment by caarlos0** [comment]
Implementation looks OK, but needs merge with main.

**Comment by saturn4er** [comment]
Done

File: env.go [hunk 1]
```diff
@@ -161,16 +164,39 @@ func optionsWithEnvPrefix(field reflect.StructField, opts Options) Options {
...
-// Parse parses a struct containing `env` tags and loads its values from
+// ParseWithOptions parses a struct containing `env` tags and loads its values from
 // environment variables.
 func ParseWithOptions(v interface{}, opts Options) error {
-	return parseInternal(v, customOptions(opts))
+	return parseInternal(v, setField, customOptions(opts))
 }
 
-func parseInternal(v interface{}, opts Options) error {
+func GetFieldParams(v interface{}) ([]FieldParams, error) {
```
Comment (caarlos0) [inline env.go:176]: missing godoc

File: env.go [hunk 2]
```diff
@@ -161,16 +164,39 @@ func optionsWithEnvPrefix(field reflect.StructField, opts Options) Options {
...
+	return GetFieldParamsWithOptions(v, defaultOptions())
+}
+
+func GetFieldParamsWithOptions(v interface{}, opts Options) ([]FieldParams, error) {
```
Comment (caarlos0) [inline env.go:180]: missing godoc

File: env.go [hunk 3]
```diff
@@ -246,71 +287,88 @@ func toEnvName(input string) string {
...
 }
 
-func get(field reflect.StructField, opts Options) (val string, err error) {
-	var exists bool
-	var isDefault bool
-	var loadFile bool
-	var unset bool
-	var notEmpty bool
-	var expand bool
+type FieldParams struct {
```
Comment (caarlos0) [inline env.go:290]: missing godoc

**Comment by caarlos0** [comment]
hey, just missing the godocs and pass `golangci-lint` :P 
sorry I missed it earlier

thank you

**Comment by saturn4er** [comment]
@caarlos0 No problem, I've made the corrections you suggested.

**Comment by caarlos0** [comment]
merged, thank you!

**Pull request merged**